components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/manager/DefaultSAML2SSOManager.java

```diff
@@ -896,28 +896,37 @@ protected RequestedAuthnContext buildRequestedAuthnContext(AuthnRequest inboundA
             requestedAuthnContext = requestedAuthnContextBuilder.buildObject();
             /* AuthnContextClass */
             AuthnContextClassRefBuilder authnContextClassRefBuilder = new AuthnContextClassRefBuilder();
-            AuthnContextClassRef authnContextClassRef = authnContextClassRefBuilder
-                    .buildObject(SAMLConstants.SAML20_NS,
-                            AuthnContextClassRef.DEFAULT_ELEMENT_LOCAL_NAME,
-                            SAMLConstants.SAML20_PREFIX);
 
             String authnContextClass = properties
                     .get(IdentityApplicationConstants.Authenticator.SAML2SSO.AUTHENTICATION_CONTEXT_CLASS);
 
             if (StringUtils.isNotEmpty(authnContextClass)) {
-                String samlAuthnContextURN = IdentityApplicationManagementUtil
-                        .getSAMLAuthnContextClasses().get(authnContextClass);
-                if (!StringUtils.isBlank(samlAuthnContextURN)) {
-                    //There was one matched URN for give authnContextClass.
-                    authnContextClassRef.setAuthnContextClassRef(samlAuthnContextURN);
-                } else {
-                    //There are no any matched URN for given authnContextClass, so added authnContextClass name to the
-                    // AuthnContextClassRef.
-                    authnContextClassRef.setAuthnContextClassRef(authnContextClass);
+                String[] authnContextClassList = authnContextClass.split(",");
+                for (String authnContextClassListElement : authnContextClassList) {
+                    AuthnContextClassRef authnContextClassRef = authnContextClassRefBuilder
+                            .buildObject(SAMLConstants.SAML20_NS,
+                                    AuthnContextClassRef.DEFAULT_ELEMENT_LOCAL_NAME,
+                                    SAMLConstants.SAML20_PREFIX);
+                    String samlAuthnContextURN = IdentityApplicationManagementUtil
+                            .getSAMLAuthnContextClasses().get(authnContextClassListElement);
+                    if (!StringUtils.isBlank(samlAuthnContextURN)) {
```

Comment:
Can we use `StringUtils.isNotBlank()` method?

Reply:
yes. resolved with this method